File: app/src/main/java/org/stepic/droid/base/SingleFragmentActivity.java
package org.stepic.droid.base;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import org.stepic.droid.R;

public abstract class SingleFragmentActivity extends FragmentActivityBase {
    protected abstract Fragment createFragment();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_fragment);

        FragmentManager fm = getSupportFragmentManager();
        Fragment fragment = fm.findFragmentById(R.id.fragmentContainer);

        if (fragment == null) {
            fragment = createFragment();
            fm.beginTransaction()
                    .add(R.id.fragmentContainer, fragment)
                    .commit();
        }

    }

}

File: app/src/main/java/org/stepic/droid/base/SingleFragmentActivity.kt
package org.stepic.droid.base

import android.os.Bundle
import androidx.fragment.app.Fragment
import org.stepic.droid.R

abstract class SingleFragmentActivity : FragmentActivityBase() {
    protected abstract fun createFragment(): Fragment
    open fun getLayoutResId() = R.layout.activity_fragment

    protected var fragment: Fragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutResId())

        val fm = supportFragmentManager
        fragment = fm.findFragmentById(R.id.fragmentContainer)


        if (fragment == null) {
            val fragment = createFragment().also { this.fragment = it }
            fm.beginTransaction()
                    .add(R.id.fragmentContainer, fragment)
//                    .addToBackStack(fragment?.javaClass?.simpleName?:"default")
                    .commit()
        }
    }

}